src/solvers/refinement/string_refinement.cpp

```diff
@@ -896,6 +896,59 @@ exprt string_refinementt::substitute_array_with_expr(
   }
 }
 
+/// Pads an array represented by a list of with_expr by propagating values to
+/// the left.
+exprt pad_array_with_expr(const exprt &expr, std::size_t string_max_length)
+{
+  PRECONDITION(expr.type().id()==ID_array);
+  mp_integer max_index=0;
+  exprt it=expr;
+
+  while(it.id()==ID_with)
```

Comment:
Instead of looping twice, probably better to simply size the vector as we go. The efficiency gain of a single allocation is probably outweighed by the risk of updating one loop but not the other in the future.